lib/TH/generic/THTensorMath.c

```diff
@@ -592,13 +639,22 @@ void THTensor_(cadd)(THTensor *r_, THTensor *t, real value, THTensor *src)
       real *sp = THTensor_(data)(src);
       real *rp = THTensor_(data)(r_);
       ptrdiff_t sz = THTensor_(nElement)(t);
-      ptrdiff_t i;
-      #pragma omp parallel for if(sz > TH_OMP_OVERHEAD_THRESHOLD) private(i)
-      for (i=0; i< sz; i++)
-          rp[i] = tp[i] + value * sp[i];
+      #pragma omp parallel if(sz > TH_OMP_OVERHEAD_THRESHOLD)
+      {
+        #ifdef _OPENMP
+        size_t num_threads = omp_get_num_threads();
+        size_t tid = omp_get_thread_num();
+        #else
+        size_t num_threads = 1;
+        size_t tid = 0;
+        #endif
+        ptrdiff_t i = tid * (sz / num_threads);
+        ptrdiff_t i_end = tid == num_threads - 1 ? sz : i + sz / num_threads;
+	THVector_(cadd)(rp+i, tp+i, sp+i, value, i_end-i); 
```

Comment:
@apaszke , Could I use `C99` flag to compile `THTensorMath.c`? As `C99` is not enabled for most files.